Issue #2053 Document available Gherkin step definitions

agajdosi opened the pull request:
Fix #2053.

agajdosi commented:
Hi @robin-owen, I have quite hard times with this issue, please take a look and see whether you think this might help contributors to better understand / get into available Gherkin steps. Many thanks.

robin-owen commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,31 @@ This information is often enough to find and debug the reason for a failure.
+All available Gherkin steps are defined in file link:https://github.com/minishift/minishift/blob/master/test/integration/testsuite/testsuite.go#L75-L259[/test/integration/testsuite.go].
+If you are searching for particular step functionality, you can use following topic shortcuts:
+
+* link:https://github.com/minishift/minishift/blob/master/test/integration/testsuite/testsuite.go#L75-L81[Minishift status and profile verification]
```
Using explicit line numbers for all of these links feels fragile to me. This list will have to be maintained if anything in **_testsuite.go_** changes.

agajdosi commented:
Removed links to exact lines, added a short introduction of how steps are implemented and how to read that, to make it easier for newcomers to get into it. Let me know what you think @robin-owen, @amitkrout.

amitkrout commented:
> Removed links to exact lines, ...

@agajdosi I could not get it. Might be i need to build the doc and check, in the mean time can you please add some more context

agajdosi commented:
@amitkrout In previous version of the PR there were shortcuts to exact topics and where they lie in the file, however after conversation with Robin I have removed those and refactored. Should have been more clear above, sorry.

amitkrout commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+To use the step in the feature file, you need to make sure it matches with the regular expression and prepend one of the Gherkin keywords: `Given`, `When`, `Then`, `And` or `But`, for example:
+
+----
+Then executing "minishift config set" succeeds
```
Example is not correct. It won't work in normal condition due to the missing ```PROPERTY_NAME``` and ```PROPERTY_VALUE```. We have a separate implementation for setting config in the testsuite.go. I think use the same steps as above so that it will be in sync 
```
When executing "minishift start" succeeds
```

agajdosi replied:
@amitkrout Good catch, will fix it, thanks.

agajdosi commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+
+[NOTE]
+====
+For more general information about structure of Gherkin, its features, scenarios and steps, please visit link:[Gherkin Reference].
```
:+1:

agajdosi commented:
@amitkrout Changes implemented, thanks for noticing.

amitkrout commented:
Retest this please

agajdosi commented:
retest this please

agajdosi commented:
@robin-owen PR should be ok now, could you please take one more look, whether it is ok? Many thanks :)

amitkrout commented:
@agajdosi  LGTM, Will approve the pr once ci run is successful.

amitkrout commented:
Retest this please

robin-owen commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+
+[NOTE]
+====
+For more general information about structure of Gherkin, its features, scenarios and steps, please visit link:https://docs.cucumber.io/gherkin/reference[Gherkin Reference].
```
```adoc
[NOTE]
====
See the link:https://docs.cucumber.io/gherkin/reference[Gherkin Reference] for more general information about the structure of Gherkin, its features, scenarios, and steps.
====
```

robin-owen commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+[[writing-features]]
+==== Writing features
+
+Minishift integration testsuite offers comprehensive set of implemented Gherkin steps which allow for comfortable writing of feature files.
```
"offers a comprehensive set"

robin-owen commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+==== Writing features
+
+Minishift integration testsuite offers comprehensive set of implemented Gherkin steps which allow for comfortable writing of feature files.
+All available steps are defined in file link:https://github.com/minishift/minishift/blob/master/test/integration/testsuite/testsuite.go[/test/integration/testsuite.go].
```
"are defined in the file"

robin-owen commented on docs/source/contributing/developing.adoc:
```diff
@@ -370,6 +371,30 @@ This information is often enough to find and debug the reason for a failure.
+    ExecutingMinishiftCommandSucceedsOrFails)
+----
+
+defines a step which matches regular expression `^executing "minishift (.*)" (succeeds|fails)$`. If matched the two capturing groups are passed as parameters to the function `ExecutingMinishiftCommandSucceedsOrFails(command string, expectedResult string)`, which implements the actual behaviour of the step.
```
"This defines a step which matches the regular expression"

"If matched, the two" (Add a comma.)

Also, please add a newline between sentences here. :)

agajdosi commented:
Thanks @robin-owen for the feedback, changes are implemented. Seems we it is ready to merge, @LalatenduMohanty.
